_posts/2017-10-05-apprenticeship-at-8thlight.md

```diff
@@ -0,0 +1,213 @@
+
+At the beginning of an apprenticeship, at the end and also while it is running the apprentices are asked to do a self-assessment on all the topics that the apprenticeship will cover.
+
+## The Curriculum
+
+As described above the curriculum contains all things about “programming, client services, collaboration, leadership, and communication skills”. The curriculum itself is way more detailed than this and contains the topics in detail, (chapters of) books to read, blog posts, tasks to solve and so on.
+
+One thing I saw that everyone at 8th Light has as part of their apprenticeship is to learn the editor vi and the terminal manager tmux. Other tools are allowed to use too, but as one part of the curriculum those two tools are required to learn. Another topic that appears in the curriculum is Test Driven Development. I remember Mollie saying that the “disciplined TDD” was one of the most important things she had learned in her apprenticeship. On [her blog][tests-that-test-you] Katerina writes about about testing on the third day of her apprenticeship. This makes obvious that testing is not added into the game at some later stage to validate something that was implemented. It is an essential part, a building block of an apprenticeship. I heard very often that the book [TDD by Example] was mentioned and is part of an apprenticeship.
+
+Another part of the curriculum is to learn the SOLID principles, open-close principle, the four rules of simple design, refactoring, coupling and cohesion (using the two chapters of the book “Structured Design”) and doing lots of Katas. The most famous kata everyone in 8th Light does is Tic Tac Toe. But besides that I saw the bowling kata, roman numerals and gilded rose. So there is a lot of practicing going on.
+
+We also talked to mentors that said they are using other projects like a contacts app, song list or building an HTTP server from scratch as projects for the apprentices. Also internal tools are often built by apprentices. 8th Light has a slack bot for collecting all food orders from all colleagues, the kanban board app and others have been built by apprentices.
+
+[TDD by Example]: https://www.goodreads.com/book/show/387190.Test_Driven_Development
+[tests-that-test-you]: https://medium.com/@katerinaldg/tests-that-test-you-1106db1cad62
+
+Part of the curriculum is also to learn three programming languages with each a different approach. The one functional: Clojure, a statically typed one: Java and a dynamic language: Ruby. In each language the apprentices normally write Tic Tac Toe first. Though this seems not to be executed that strictly anymore.
```

Comment:
Those are the traditional languages, but of late Elixir has been replacing Clojure in some cases, and C# instead of java. We look at the apprentice's experience to help guide the decision of which languages to use (if they've used java day-to-day, we'll do c# instead).